File: 백준/Silver/1940. 주몽/주몽.py
import sys

input = sys.stdin.readline
print = sys.stdout.write

n = int(input())
m = int(input())

arr = list(map(int, input().split()))

arr.sort()

s = 0
e = n-1
cnt = 0

while s != e:
    sum = arr[s] + arr[e]
    if (sum == m):
        cnt += 1
        e -= 1
    elif (sum > m):
        e -= 1
    else:
        s += 1
        
        
print(str(cnt))

File: 백준/Silver/1940. 주몽/주몽.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        
        int n = Integer.parseInt(br.readLine());
        int m = Integer.parseInt(br.readLine());
        
        StringTokenizer st = new StringTokenizer(br.readLine());
        
        int[] arr = new int[n];
        
        for (int i = 0; i < n; i++) {
            arr[i] = Integer.parseInt(st.nextToken());
        }
        
        Arrays.sort(arr);
        
        int s = 0;
        int e = n-1;
        int cnt = 0;
        
        while(s != e) {
            int sum = arr[s] + arr[e];
            if (sum == m) {
                cnt++;
                e--;
            } else if (sum > m) {
                e--;
            } else {
                s++;
            }
        }
        
        bw.write(cnt + "\n");
        bw.flush();
        bw.close();
    }
}